packages/router/router.ts

```diff
@@ -1455,13 +1485,16 @@ export function createRouter(init: RouterInit): Router {
     let matches = matchRoutes(routesToUse, location, basename);
     let flushSync = (opts && opts.flushSync) === true;
 
+    let fogOfWar = checkFogOfWar(matches, routesToUse, location.pathname);
+    if (fogOfWar.active && fogOfWar.matches) {
+      matches = fogOfWar.matches;
+    }
```

Comment:
Detect if this is a fog of war scenario and we should bypass a synchronous 404 and go into our loading/submitting state to perform discovery